ZipTest/Program.cs

```diff
@@ -63,12 +65,123 @@ public static void Main (string [] args)
 				}
 			}
 
+
+			if (File.Exists ("test-archive-write.zip"))
+				File.Delete ("test-archive-write.zip");
+
+			var t = "sdfjha;ouwrhewourfh;eajfbeouwfbdjabfkljdsbfakjsbfkjadsf";
+
+			var ms = new MemoryStream (Encoding.UTF8.GetBytes (t));
+
 			string asmPath = typeof (MainClass).GetType ().Assembly.Location;
 			using (var zip = ZipArchive.Open ("test-archive-write.zip", FileMode.CreateNew)) {
 				zip.AddFile (asmPath);
 				zip.AddFile (asmPath, "/in/archive/path/ZipTestCopy.exe");
-				zip.AddFile (asmPath, "/in/archive/path/ZipTestCopy2.exe/", permissions: EntryPermissions.OwnerRead | EntryPermissions.OwnerWrite);
+				zip.AddFile (asmPath, "/in/archive/path/ZipTestCopy2.exe", permissions: EntryPermissions.OwnerRead | EntryPermissions.OwnerWrite);
+				zip.AddFile (asmPath, "/in/archive/path/ZipTestCopy3.exe", method: CompressionMethod.STORE);
+				var text = "Hello World";
+				zip.AddEntry ("/in/archive/data/foo.txt", text, Encoding.UTF8, CompressionMethod.STORE);
+
+				using (var fs = File.OpenRead (asmPath)) {
+					zip.AddEntry ("/in/archive/foo/foo.exe", fs, CompressionMethod.STORE);
+				}
+				zip.AddStream (ms, "/in/archive/foo/foo1.txt", method: CompressionMethod.STORE);
+			}
+
+			if (File.Exists ("test-archive-write.zip")) {
+				using (var zip = ZipArchive.Open (File.OpenRead ("test-archive-write.zip"))) {
+					foreach (var e in zip) {
+						Console.WriteLine ($" {e.Name} {e.Size} {e.CompressedSize} {e.CompressionMethod}");
+					}
+				}
+			}
+		}
+	}
+
+	public enum CompressionLevel
+	{
+		NoCompression, 
+		Optimal,
+		Fastest,
+	}
+
+	static class ZipArchiveExtensions
```

Comment:
You're right that those extensions should be added to the API. Can you put them directly in ZipArchive?